framework/src/org/apache/cordova/engine/SystemWebChromeClient.java

```diff
@@ -211,54 +216,137 @@ public View getVideoLoadingProgressView() {
         return mVideoProgressView;
     }
 
+    // <input type=file> support:
+    // openFileChooser() is for pre KitKat and in KitKat mr1 (it's known broken in
+    // KitKat).
+    // For Lollipop, we use onShowFileChooser().
```

Comment:
TODO: double check Cordova supported versions, I'm 99% sure we've dropped KitKat support already, and I'm pretty sure Lollipop is also out (or will be out in the upcoming major version bump)

Reply:
Kitkat (Android 4.4 / API 19) was indeed dropped since cordova-android@10.
Lollipop (Android 5.0 / API 21) is being dropped in the next release in cordova-android@12, which will hvae a Min SDK of 24 (Android 7 / Nougat)

But I assume [onShowFileChooser](https://developer.android.com/reference/android/webkit/WebChromeClient#onShowFileChooser(android.webkit.WebView,%20android.webkit.ValueCallback%3Candroid.net.Uri[]%3E,%20android.webkit.WebChromeClient.FileChooserParams)) is to be used for API 21+, so I think it's still relevant. It's not deprecated.

Reply:
Since Cordova-Android 9.0 the `minSdkVersion` is actually set to SDK 22 (Android 5.1) Lollipop MR1, not API 21.

Reply:
Nice.  Then we can just remove all this code.  I will delete it.